src/components/SubPage/EditingDesignSubpage.tsx

```diff
@@ -0,0 +1,85 @@
+import React from "react";
+import SubpageContent from "./Subpage";
+import { Diagram } from "../../types";
+
+const {
+    pamSite,
+    mutation,
+} = require("../../style/editing-design.module.css");
+
+interface EditingDesignContentProps {
+    editingDesignData: {
+        crna_target_site: string;
+        dna_donor_sequence: string;
+        cas9: string;
+        f_primer: string;
+        r_primer: string;
+        diagrams: Diagram[];
+    };
+}
+
+
+const formatTextWithGeneLocations = (text: string, className: string) => {
+    // PAM sites and mutations are indicated in the string using square brackets
+    const parts = text.split(/(\[.*?\])/);
+    return parts.map((part, index) => {
+        if (part.startsWith("[") && part.endsWith("]")) {
+            return (
+                <span key={index} className={className}>
+                    {part}
+                </span>
+            );
+        }
+        return part;
+    });
+};
+
+const EditingDesignContent: React.FC<EditingDesignContentProps> = ({
+    editingDesignData,
+}) => {
+    const contentData = React.useMemo(
```

Comment:
this doesn't need a useMemo: it's the only prop, it'll re-render every time the prop changes, so you're not saving any render cycles here unless I'm missing something